File: composeApp/src/toDestroyJava/src/Enumerise/Reflevel.java
package Enumerise;

import java.util.HashMap;
import java.util.Map;

public enum Reflevel {
MIN("MIN",0),
MAX("MAX",1),
OPTIMIN("OPTIMIN",2),

OPTIMAX("OPTIMAX",3)
			;
			


		private String name = "";
		private int coef = 0;

		//Constructeur
		Reflevel(String name, int coef){
		this.name = name;
		this.coef=coef;
		}
		public static Reflevel getById(int i){
			
			return (Reflevel) map.get(i);
		}
		 private static Map map = new HashMap<>();
		static {
	        for (Reflevel pageType : Reflevel.values()) {
	            map.put(pageType.coef, pageType);
	        }
	    }

		public String nameToString(){
		return name;
		}
		public int getCoef(){
			  return coef;
			}
		public static int size(){
			  return 4;
			}
		}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/Reflevel.kt
package fr.vetbrain.vetnutri_mp.Enumer

import fr.vetbrain.vetnutri_mp.Data.Labelable

/**
 * Énumération représentant les niveaux relatifs des références de nutriments Basée sur
 * l'énumération Reflevel du projet Java original
 */
enum class Reflevel(val uuid: Int, override val label: String) : Labelable {
    MIN(0, "minimum"),
    MAX(1, "maximum"),
    OPTIMIN(2, "optimumMin"),
    OPTIMAX(3, "optimumMax");

    companion object {
        /**
         * Récupère un niveau relatif par son identifiant
         *
         * @param id L'identifiant à rechercher
         * @return Le niveau relatif correspondant ou MIN par défaut
         */
        fun getById(id: Int): Reflevel = entries.find { it.uuid == id } ?: MIN
    }

    override fun toString(): String = label
}
